summary/summary.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+from lightning import Plugin, Millisatoshi
+import json
+import requests
+import threading
+import time
+
+plugin = Plugin(autopatch=True)
+
+
+class PriceThread(threading.Thread):
+    def run(self):
+        try:
+            r = requests.get('https://apiv2.bitcoinaverage.com/convert/global'
+                             '?from=BTC&to={}&amount=1'.format(plugin.currency))
+            plugin.fiat_per_btc = json.loads(r.content)['price']
+        except Exception:
+            pass
+        # Six hours is more than often enough for polling
+        time.sleep(6*3600)
+
+
+def to_fiatstr(msat: Millisatoshi):
+    return "{}{:.2f}".format(plugin.currency_prefix,
+                              int(msat) / 10**11 * plugin.fiat_per_btc)
+
+
+@plugin.method("summary")
+def summary(plugin):
+    """Gets summary information about this node."""
+
+    reply = {}
+    info = plugin.rpc.getinfo()
+    funds = plugin.rpc.listfunds()
+    peers = plugin.rpc.listpeers()
+
+    # Make it stand out if we're not on mainnet.
+    if info['network'] != 'bitcoin':
+        reply['network'] = info['network'].upper()
+
+    if not plugin.my_address:
+        reply['warning_no_address'] = "NO PUBLIC ADDRESSES"
+    else:
+        reply['my_address'] = plugin.my_address
+
+    utxo_amount = Millisatoshi(0)
+    reply['num_utxos'] = 0
```

Comment:
List comprehension might be nice here, and then `sum` and `len` of that list :-)

Reply:
Hmm, not sure it's simpler, but it is shorter...